### ISSUE TITLE
Create Item If doesn't Exist

### ISSUE THREAD

**Issue by shoaibshakeel381** [arendjr/selectivity#39]
How can i configure `selectivity` to allow user to enter a value in search input and if the item doesn't exist it should persist user input as new option. Like in `selectize.js` when an item doesn't exist it it allows user to add current entered input as new option. [Demo Page](http://brianreavis.github.io/selectize.js/)

**Comment by arendjr** [comment]
Are you looking for the `createTokenItem` item fuction? Are you working with a multi-value or a single-value input? Because `createTokenItem` is only used with multiple-value inputs so far, so maybe your request would be to make it possible to use with single-value inputs as well :)

**Comment by shoaibshakeel381** [comment]
I am using single-value input. Currently i am using a hack to accomplish my requirements. But it would be best if `selectivity` supported if itself. Thnx

**Comment by cardi777** [comment]
arendjr - i see you added the enhancement - pardon my lack of know-how, but you are referring to making `createTokenItem` work in single item mode? im trying to accomplish this as well, love the plugin!

**Comment by arendjr** [comment]
Yes, that's pretty much what it means. I think other libraries also refer to it as "tagging" mode. I've updated the title to make it more clear.

**Comment by alexcroox** [comment]
Is there any examples on how to implement createTokenItem, the documentation on it is very light.
